tests/basic_sanity_test.go

```diff
@@ -15,7 +15,10 @@ const (
 )
 
 var _ = Describe(TestSuiteName, func() {
-	f := framework.NewFramework("sanity")
+	c := framework.FrameworkConfig {
+		SkipNamespaceCreation: true,
+	}
+	f := c.NewFramework("sanity")
```

Comment:
another option is to just pass the config in as a separate parameter to NewFrameWork.

```
c.newFramework("sanity", &framework.FrameworkConfig{})
```

and 

```
c.newFramework("sanity", &framework.FrameworkConfig{SkipNamespaceCreation: true})```

```

Reply:
Right, I have seen 3 different way people pass in config/options at this point:

1. Expose them in the struct, and let the caller set them.
2. Pass the config struct into the newXXX (like what @davidvossel is suggesting)
3. Create the config, and then have the config create the instance of the struct (like what I am doing here).

I am trying to do it the correct golang way as much as possible. I have seen all 3 in examples I looked at. What do we normally do in kubevirt?

Reply:
in k8s I've seen the pattern where we pass 'options' as an additional parameter to a function.

For example.
```
clientset.CoreV1.Pods(namespace).Delete(podName, &v1.DeleteOptions{})
```

In the event that no deletion options are necessary, it's pretty simple to just add an empty object in that function call. When deletion options are necessary, it's still a single line of code. The advantage here is that as more options are added to the DeletionOptions struct, the Delete() function's interface doesn't need to change.

Reply:
@davidvossel I have seen that as well, but I have also seen this [writer](https://github.com/ulikunitz/xz/blob/636d36a76670e6c700f22fd5f4588679ff2896c4/writer.go#L174) as pointed out by @jeffvance. I have no strong opinion on any of the 3 options. But before I make more changes, I would like consensus on which approach we would like to use.

Reply:
@awels thanks for considering the 3 ways of presenting config options to the Framework constructor.! Although I think the xz Writer is clever, I agree with David that an options struct as an additional parm is fairly typical in k8s. I'd go with that approach.